mlir-tensorrt/executor/include/mlir-executor-c/Runtime/Runtime.h

```diff
@@ -375,15 +386,22 @@ static inline bool mtrtRuntimeSessionIsNull(MTRT_RuntimeSession session) {
 }
 
 /// Using `session`, execute the pubic function with the specified name.
-/// The `inArgs` and `outArgs` are arrays for input arguments and destination
-/// arguments, respectively. Input arguments may be MemRefs or scalars, but
-/// destination arguments must be MemRefs.
-/// A stream may optionally be specified, otherwise pass the result of
-/// `mtrtStreamGetNull()`.
+/// The `inArgs`, `outArgs`, and `results` are arrays for input arguments and
+/// optional destination arguments and results, respectively. Input arguments
+/// may be MemRefs or scalars, but destination arguments and results must be
+/// MemRefs. If `outArgs` are present, it is expected that `results` are empty
+/// and vice-versa. A stream may optionally be specified, otherwise pass the
+/// result of `mtrtStreamGetNull()`.
 MLIR_CAPI_EXPORTED MTRT_Status mtrtRuntimeSessionExecuteFunction(
     MTRT_RuntimeSession session, MTRT_StringView name,
     const MTRT_RuntimeValue *inArgs, size_t numInArgs,
-    const MTRT_RuntimeValue *outArgs, size_t numOutArgs, MTRT_Stream stream);
+    const MTRT_RuntimeValue *outArgs, size_t numOutArgs,
+    MTRT_RuntimeValue *results, size_t numResults, MTRT_Stream stream,
```

Comment:
It shouldn't be necessary to pass `numResults` here. You can just say that `results` must point to an array of minimum size given by `mtrtRuntimeSessionGetNumResults`.